cmd/clair/main.go

```diff
@@ -102,40 +102,13 @@ func stopCPUProfiling(f *os.File) {
 }
 
 func configClairVersion(config *Config) {
-	listers := featurefmt.ListListers()
-	detectors := featurens.ListDetectors()
-	updaters := vulnsrc.ListUpdaters()
+	clair.EnabledDetectors = append(featurefmt.ListListers(), featurens.ListDetectors()...)
+	clair.EnabledUpdaters = strutil.Intersect(config.Updater.EnabledUpdaters, vulnsrc.ListUpdaters())
 
 	log.WithFields(log.Fields{
-		"Listers":   strings.Join(listers, ","),
-		"Detectors": strings.Join(detectors, ","),
-		"Updaters":  strings.Join(updaters, ","),
-	}).Info("Clair registered components")
-
-	unregDetectors := strutil.CompareStringLists(config.Worker.EnabledDetectors, detectors)
-	unregListers := strutil.CompareStringLists(config.Worker.EnabledListers, listers)
-	unregUpdaters := strutil.CompareStringLists(config.Updater.EnabledUpdaters, updaters)
-	if len(unregDetectors) != 0 || len(unregListers) != 0 || len(unregUpdaters) != 0 {
-		log.WithFields(log.Fields{
-			"Unknown Detectors":   strings.Join(unregDetectors, ","),
-			"Unknown Listers":     strings.Join(unregListers, ","),
-			"Unknown Updaters":    strings.Join(unregUpdaters, ","),
-			"Available Listers":   strings.Join(featurefmt.ListListers(), ","),
-			"Available Detectors": strings.Join(featurens.ListDetectors(), ","),
-			"Available Updaters":  strings.Join(vulnsrc.ListUpdaters(), ","),
-		}).Fatal("Unknown or unregistered components are configured")
-	}
-
-	// verify the user specified detectors/listers/updaters are implemented. If
-	// some are not registered, it logs warning and won't use the unregistered
-	// extensions.
-
-	clair.Processors = database.Processors{
-		Detectors: strutil.CompareStringListsInBoth(config.Worker.EnabledDetectors, detectors),
-		Listers:   strutil.CompareStringListsInBoth(config.Worker.EnabledListers, listers),
-	}
-
-	clair.EnabledUpdaters = strutil.CompareStringListsInBoth(config.Updater.EnabledUpdaters, updaters)
+		"Detectors": strings.Join(database.SerializeDetectors(clair.EnabledDetectors), ","),
+		"Updaters":  strings.Join(clair.EnabledUpdaters, ","),
```

Comment:
Do these have to be strings? I'd be nice to keep them as arrays in the JSON logs.
Also the detectors string is borderline unreadable:

```json
{"Detectors":"DETECTOR_TYPE_FEATUREDetector/apk/1.0,DETECTOR_TYPE_FEATUREDetector/dpkg/1.0,DETECTOR_TYPE_FEATUREDetector/rpm/1.0,DETECTOR_TYPE_NAMESPACEDetector/alpine-release/1.0,DETECTOR_TYPE_NAMESPACEDetector/apt-sources/1.0,DETECTOR_TYPE_NAMESPACEDetector/lsb-release/1.0,DETECTOR_TYPE_NAMESPACEDetector/os-release/1.0,DETECTOR_TYPE_NAMESPACEDetector/redhat-release/1.0"}
```